starlark/debug.go

```diff
@@ -0,0 +1,18 @@
+package starlark
+
+// This file defines an experimental API for the debugging tools.
+// Some of these declarations expose details of internal packages.
+// (The debugger makes liberal use of exported fields of unexported types.)
+// Breaking changes may occur without notice.
+
+// Local returns the value of the i'th local variable.
+// It may be nil if not yet assigned.
+//
+// Local may be called only for frames whose Callable is a *Function (a
+// function defined by Starlark source code), and only while the frame
+// is active; it will panic otherwise.
+//
+// This function is provided only for debugging tools.
+//
+// THIS API IS EXPERIMENTAL AND MAY CHANGE WITHOUT NOTICE.
```

Comment:
Since users are encouraged not to call this, I wonder if it would be better to name this `DebugLocal`? People might not read the docs.

Reply:
There are going to be a number of functions that are intended for debugger-like tools and other diagnostic machinery but that shouldn't generally be used to influence the behavior of a Starlark program. I don't want to have to have to put Debug in all their names. Most clients will have little reason to find themselves using this; the comment is fair warning to "advanced" aka "overly curious" users.